packages/marble-datatable/src/Datatable.js

```diff
@@ -345,13 +345,23 @@ class Datatable extends Component {
     return core.isObject(data) && 'columns' in data && 'type' in data;
   }
 
+  /**
+   * Clone JSON data
+   * @param {*} data
+   * @return {*}
+   */
+  cloneJSONData(data) {
+      return JSON.parse(JSON.stringify(data));
+  }
+
   /**
    * Setter for the `data` state property.
    * @param {!Object}
    * @return {!Object}
    * @protected
    */
-  setData_(data) {
+  setData_(givenData) {
```

Comment:
I got some errors on JSON stringify and parse when testing locally because sometimes the `givenData` is undefined. I think you could add the default value here to prevent those errors if that makes sense. Like this:

```javascript
setData_(givenData = {}) {
```

Reply:
I opted to handle it in the `cloneJSONData` function, changing it to `maybeCloneJSONData`, this way we don't make the parameter opinionated as to type